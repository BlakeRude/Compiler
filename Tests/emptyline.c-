// C-S21
main();

dogs()
{
    int x;
    ;
    ;
    if true then ;
    else ;
    ;
    while true do ;
    for x=1 to 10 do ;
    {}
}
