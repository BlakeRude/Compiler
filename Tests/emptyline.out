====================================
FILE: emptyline.c-
WARNING(13): The variable 'x' seems not to be used.
WARNING(6): The variable 'x' seems not to be used.
WARNING(4): The function 'dogs' seems not to be used.
Func: main returns type void [line: 2]
Sibling: 1  Func: dogs returns type void [line: 4]
.   Child: 1  Compound [line: 5]
.   .   Child: 0  Var: x of type int [line: 6]
.   .   Child: 1  If [line: 9]
.   .   .   Child: 0  Const true of type bool [line: 9]
.   .   Sibling: 1  While [line: 12]
.   .   .   Child: 0  Const true of type bool [line: 12]
.   .   Sibling: 2  For [line: 13]
.   .   .   Child: 0  Var: x of type int [line: 13]
.   .   .   Child: 1  Range [line: 13]
.   .   .   .   Child: 0  Const 1 of type int [line: 13]
.   .   .   .   Child: 1  Const 10 of type int [line: 13]
.   .   Sibling: 3  Compound [line: 14]
Number of warnings: 3
Number of errors: 0
